packages/hasher/src/workspaces/getWorkspaces.ts

```diff
@@ -1,23 +1,38 @@
-import findUp from "find-up";
+import { fastFindUp } from "./fastFindUp";
 import { getPnpmWorkspaces } from "./pnpmWorkspaces";
 import { getYarnWorkspaces } from "./yarnWorkspaces";
 import { getRushWorkspaces } from "./rushWorkspaces";
 import { WorkspaceInfo } from "./WorkspaceInfo";
 
-export function getWorkspaces(cwd: string): WorkspaceInfo {
-  const yarnLockPath = findUp.sync("yarn.lock", { cwd });
+const workspaceCache = new Map<string, WorkspaceInfo>();
+
+/**
+ * Gets workspace packages, get cached based on the kind of workspace
+ * @param cwd
+ */
+export async function getWorkspaces(cwd: string): Promise<WorkspaceInfo> {
+  const yarnLockPath = await fastFindUp("yarn.lock", cwd);
   if (yarnLockPath) {
-    return getYarnWorkspaces(cwd);
+    if (!workspaceCache.has("yarn")) {
+      workspaceCache.set("yarn", getYarnWorkspaces(cwd));
+    }
+    return workspaceCache.get("yarn")!;
```

Comment:
Similar to my comment about `fastFindUp`, maybe we could use `mem` here?

Reply:
i think so - I've not heard of mem - assuming it's memoize kind of thing

Reply:
Okay, this usage of fastFindUp is gone :)